Wox.Plugin.Runner/Runner.cs

```diff
@@ -1,163 +1,165 @@
-                    throw;
-            }
-            catch (FormatException)
-            {
-                Context.API.ShowMsg("There was a problem. Please check the arguments format for the command.");
-            }
-            return true;
-        }
-
-        private ProcessArguments GetProcessArguments(Command c, IEnumerable<string> terms)
-        {
-            var argString = string.Empty;
-
-            if (!string.IsNullOrEmpty(c.ArgumentsFormat))
-            {
-                if (c.ArgumentsFormat.EndsWith("{*}"))
-                {
-                    argString = c.ArgumentsFormat.Remove(c.ArgumentsFormat.Length - 3, 3) + string.Join(" ", terms);
-                }
-                else
-                {
-                    argString = string.Format(c.ArgumentsFormat, terms.ToArray());
-                }
-            }
-
-            var workingDir = c.WorkingDirectory;
-            if (string.IsNullOrEmpty(workingDir))
-            {
-                // Use directory where executable is based.
-                workingDir = Path.GetDirectoryName(c.Path);
-            }
-            return new ProcessArguments
-            {
-                FileName = c.Path,
-                Arguments = argString,
-                WorkingDirectory = workingDir
-            };
-        }
-
-        class ProcessArguments
-        {
-            public string FileName { get; set; }
-            public string Arguments { get; set; }
-            public string WorkingDirectory { get; set; }
-        }
-    }
+using Flow.Launcher.Plugin;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Windows.Controls;
+using Wox.Plugin.Runner.Settings;
+
+namespace Wox.Plugin.Runner
+{
+    public class Runner : IPlugin, ISettingProvider
+    {
+        internal static PluginInitContext Context;
+        RunnerSettingsViewModel viewModel;
+
+        public void Init(PluginInitContext context)
+        {
+            Context = context;
+            viewModel = new RunnerSettingsViewModel(Context);
+        }
+
+        public List<Result> Query(Query query)
+        {
+            var results = new List<Result>();
+
+            var search = query.Search;
+
+            // triggers when an action keyword is set
+            // shows all possible plugin commands
+            if (string.IsNullOrEmpty(search))
+            {
+                results = RunnerConfiguration.Commands
+                    .Select(c =>
+                        new Result()
+                        {
+                            Score = 50,
+                            Title = c.Shortcut,
+                            SubTitle = c.Description,
+                            Action = e =>
+                            {
+                                var splittedSearch = c.Shortcut.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Hey thanks for the PR. All seems ok except for this part, which seems to allow using Shortcut to specify argument, what's the intention behind this?

Reply:
I don't recall having having any specific intension of allowing using a command's Shortcut to specify arguments. I apologize, like I said originally C# isn't a language I'm comfortable with. I believe I was using the following line from the original code
`var splittedSearch = search.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);`
for reference when I made that change.

Upon review and testing I now see how that change has the unintended side effect of passing along a command's Shortcut as an argument. I've come up with a fix that involves making the `terms` parameter optional.

Reply:
Hey all good, I have merged the PR. I have additionally made a fix with the if else statement and also updated readme to clarify how to use the arguments. If you think further readability improvement can be done or more enhancements please feel free to put in more changes :)